mullvad-relay-selector/src/relay_selector/matcher.rs

```diff
@@ -38,10 +39,10 @@ pub fn filter_matching_relay_list(
             .filter(|relay| filter_on_ownership(&query.ownership, relay))
             // Filter by providers
             .filter(|relay| filter_on_providers(&query.providers, relay))
-            // Filter by DAITA support
-            .filter(|relay| filter_on_daita(&query.wireguard_constraints.daita, relay))
             // Filter by obfuscation support
-            .filter(|relay| filter_on_obfuscation(&query.wireguard_constraints, relay_list, relay));
+            .filter(|relay| filter_on_obfuscation(&query.wireguard_constraints, relay_list, relay))
+            // Filter on core privacy features
+            .filter(|relay| filter_on_core_privacy_features(&query.wireguard_constraints, relay));
```

Comment:
I am a little bit concerned about this design deviating from what is the intent of the relay query, since there's much more logic in the filtering step than usual. I think that ideally you should be able to see that a query with only openvpn and a query with e.g. PQ are incompatible when they are intersected, without having to parse the relay list. It may be hard to make e.g. queries with "only openVPN" and "only Wireguard" not representable, but perhaps we could make them impossible to construct? What do you think about modifying the relay query builder and the impl<'a> From<NormalSelectorConfig<'a>> for RelayQuery block so that tunnel_protocol is set to Only(WireGuard) whenever a privacy feature is enabled, and possibly removing filter_on_core_privacy_features entirely?

Reply:
Fixed by making it impossible to construct invalid queries (as discussed).